### PR TITLE
Documentation improvements

### PR DESCRIPTION
**Pull request by calebeby**
The goal with the long lists of methods is to make stuff more searchable, so if a user comes to the README and does "find in page" looking for "screenshot" or "toBeEnabled" or "queryAllByRole" or something they get useful results

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -648,15 +658,19 @@ test(
...
...ired), [`toBeValid`](https://github.com/testing-library/jest-dom#tobevalid), [`toBeVisible`](https://github.com/testing-library/jest-dom#tobevisible), [`toContainElement`](https://github.com/testing-library/jest-dom#tocontainelement), [`toContainHTML`](https://github.com/testing-library/jest-dom#tocontainhtml), [`toHaveAttribute`](https://github.com/testing-library/jest-dom#tohaveattribute), [`toHaveClass`](https://github.com/testing-library/jest-dom#tohaveclass), [`toHaveFocus`](https://github.com/testing-library/jest-dom#tohavefocus), [`toHaveFormValues`](https://github.com/testing-library/jest-dom#tohaveformvalues), [`toHaveStyle`](https://github.com/testing-library/jest-dom#tohavestyle), [`toHaveTextContent`](https://github.com/testing-library/jest-dom#tohavetextcontent), [`toHaveValue`](https://github.com/testing-library/jest-dom#tohavevalue), [`toHaveDisplayValue`](https://github.com/testing-library/jest-dom#tohavedisplayvalue), [`toBeChecked`](https://github.com/testing-library/jest-dom#tobechecked), [`toBePartiallyChecked`](https://github.com/testing-library/jest-dom#tobepartiallychecked), [`toHaveDescription`](https://github.com/testing-library/jest-dom#tohavedescription).
```
Comment (spaceninja) [inline README.md:663]:
```suggestion
- [`toBeDisabled`](https://github.com/testing-library/jest-dom#tobedisabled)
- [`toBeEnabled`](https://github.com/testing-library/jest-dom#tobeenabled)
- [`toBeEmpty`](https://github.com/testing-library/jest-dom#tobeempty)
- [`toBeEmptyDOMElement`](https://github.com/testing-library/jest-dom#tobeemptydomelement)
- [`toBeInTheDocument`](https://github.com/testing-library/jest-dom#tobeinthedocument)
- [`toBeInvalid`](https://github.com/testing-library/jest-dom#tobeinvalid)
- [`toBeRequired`](https://github.com/testing-library/jest-dom#toberequired)
- [`toBeValid`](https://github.com/testing-library/jest-dom#tobevalid)
- [`toBeVisible`](https://github.com/testing-library/jest-dom#tobevisible)
- [`toContainElement`](https://github.com/testing-library/jest-dom#tocontainelement)
- [`toContainHTML`](https://github.com/testing-library/jest-dom#tocontainhtml)
- [`toHaveAttribute`](https://github.com/testing-library/jest-dom#tohaveattribute)
- [`toHaveClass`](https://github.com/testing-library/jest-dom#tohaveclass)
- [`toHaveFocus`](https://github.com/testing-library/jest-dom#tohavefocus)
- [`toHaveFormValues`](https://github.com/testing-library/jest-dom#tohaveformvalues)
- [`toHaveStyle`](https://github.com/testing-library/jest-dom#tohavestyle)
- [`toHaveTextContent`](https://github.com/testing-library/jest-dom#tohavetextcontent)
- [`toHaveValue`](https://github.com/testing-library/jest-dom#tohavevalue)
- [`toHaveDisplayValue`](https://github.com/testing-library/jest-dom#tohavedisplayvalue)
- [`toBeChecked`](https://github.com/testing-library/jest-dom#tobechecked)
- [`toBePartiallyChecked`](https://github.com/testing-library/jest-dom#tobepartiallychecked)
- [`toHaveDescription`](https://github.com/testing-library/jest-dom#tohavedescription)
```

File: README.md [hunk 2]
```diff
@@ -665,6 +679,60 @@ test(
...
...#?product=Puppeteer&version=v9.1.1&show=api-pagesetgeolocationoptions), [`page.setOfflineMode`](https://pptr.dev/#?product=Puppeteer&version=v9.1.1&show=api-pagesetofflinemodeenabled), [`page.setRequestInterception`](https://pptr.dev/#?product=Puppeteer&version=v9.1.1&show=api-pagesetrequestinterceptionvalue-cachesafe), [`page.title`](https://pptr.dev/#?product=Puppeteer&version=v9.1.1&show=api-pagetitle), [`page.url`](https://pptr.dev/#?product=Puppeteer&version=v9.1.1&show=api-pageurl), [`page.waitForNavigation`](https://pptr.dev/#?product=Puppeteer&version=v9.1.1&show=api-pagewaitfornavigationoptions), [`page.browserContext().overridePermissions`](https://pptr.dev/#?product=Puppeteer&version=v9.1.1&show=api-browsercontextoverridepermissionsorigin-permissions), [`page.keyboard.press`](https://pptr.dev/#?product=Puppeteer&version=v9.1.1&show=api-keyboardpresskey-options), [`page.mouse.move`](https://pptr.dev/#?product=Puppeteer&version=v9.1.1&show=api-mousemovex-y-options), [`page.mouse.click`](https://pptr.dev/#?product=Puppeteer&version=v9.1.1&show=api-mouseclickx-y-options), [`page.touchscreen.tap`](https://pptr.dev/#?product=Puppeteer&version=v9.1.1&show=api-touchscreentapx-y)
```
Comment (spaceninja) [inline README.md:734]:
```suggestion
- [`page.cookies`](https://pptr.dev/#?product=Puppeteer&version=v9.1.1&show=api-pagecookiesurls)
- [`page.emulateMediaFeatures`](https://pptr.dev/#?product=Puppeteer&version=v9.1.1&show=api-pageemulatemediafeaturesfeatures)
- [`page.emulateNetworkConditions`](https://pptr.dev/#?product=Puppeteer&version=v9.1.1&show=api-pageemulatenetworkconditionsnetworkconditions)
- [`page.evaluate`](https://pptr.dev/#?product=Puppeteer&version=v9.1.1&show=api-pageevaluatepagefunction-args)
- [`page.evaluateHandle`](https://pptr.dev/#?product=Puppeteer&version=v9.1.1&show=api-pageevaluatehandlepagefunction-args)
- [`page.exposeFunction`](https://pptr.dev/#?product=Puppeteer&version=v9.1.1&show=api-pageexposefunctionname-puppeteerfunction)
- [`page.goBack`](https://pptr.dev/#?product=Puppeteer&version=v9.1.1&show=api-pagegobackoptions)
- [`page.goForward`](https://pptr.dev/#?product=Puppeteer&version=v9.1.1&show=api-pagegoforwardoptions)
- [`page.goto`](https://pptr.dev/#?product=Puppeteer&version=v9.1.1&show=api-pagegotourl-options)
- [`page.metrics`](https://pptr.dev/#?product=Puppeteer&version=v9.1.1&show=api-pagemetrics)
- [`page.reload`](https://pptr.dev/#?product=Puppeteer&version=v9.1.1&show=api-pagereloadoptions)
- [`page.screenshot`](https://pptr.dev/#?product=Puppeteer&version=v9.1.1&show=api-pagescreenshotoptions)
- [`page.setGeolocation`](https://pptr.dev/#?product=Puppeteer&version=v9.1.1&show=api-pagesetgeolocationoptions)
- [`page.setOfflineMode`](https://pptr.dev/#?product=Puppeteer&version=v9.1.1&show=api-pagesetofflinemodeenabled)
- [`page.setRequestInterception`](https://pptr.dev/#?product=Puppeteer&version=v9.1.1&show=api-pagesetrequestinterceptionvalue-cachesafe)
- [`page.title`](https://pptr.dev/#?product=Puppeteer&version=v9.1.1&show=api-pagetitle)
- [`page.url`](https://pptr.dev/#?product=Puppeteer&version=v9.1.1&show=api-pageurl)
- [`page.waitForNavigation`](https://pptr.dev/#?product=Puppeteer&version=v9.1.1&show=api-pagewaitfornavigationoptions)
- [`page.browserContext().overridePermissions`](https://pptr.dev/#?product=Puppeteer&version=v9.1.1&show=api-browsercontextoverridepermissionsorigin-permissions)
- [`page.keyboard.press`](https://pptr.dev/#?product=Puppeteer&version=v9.1.1&show=api-keyboardpresskey-options)
- [`page.mouse.move`](https://pptr.dev/#?product=Puppeteer&version=v9.1.1&show=api-mousemovex-y-options)
- [`page.mouse.click`](https://pptr.dev/#?product=Puppeteer&version=v9.1.1&show=api-mouseclickx-y-options)
- [`page.touchscreen.tap`](https://pptr.dev/#?product=Puppeteer&version=v9.1.1&show=api-touchscreentapx-y)
```

**Review by spaceninja — APPROVED** [review]
Great changes!

I reformatted two lists into proper unordered lists: This makes the list easier to scan visually, and also makes the file easier to maintain in the future if we add/remove methods.

File: README.md (hunk 1 above)
Comment (calebeby) [reply README.md:663]: I had mostly intended to have these words here as "search targets", that you wouldn't really need to read or scan through, and really are there for the benefit of <kbd>⌘</kbd>-<kbd>f</kbd>. Having a full `<ol>` emphasizes them more than I had wanted to

**Pull request merged**